ssg/variables.py

```diff
@@ -69,277 +64,94 @@ def get_variable_files(content_dir: str) -> list_type[str]:
     return variable_files
 
 
-def get_variable_options(content_dir: str, variable_id: str = None) -> dict_type:
+def _get_variables_content(content_dir: str) -> dict_type:
     """
-    Retrieve the options for specific or all variables from the content root directory.
-
-    If `variable_id` is provided, returns options for that variable only.
-    If `variable_id` is not provided, returns a dictionary of all variable options.
+    Retrieve the content of all variable files from the specified content root directory.
 
     Args:
         content_dir (str): The root directory containing benchmark directories.
-        variable_id (str, optional): The ID of the variable to retrieve options for.
-                                     Defaults to None.
 
     Returns:
-        dict: If `variable_id` is None, a dictionary where keys are variable IDs and values are
-              their options. Otherwise, a dictionary of options for the specified variable.
+        dict: A dictionary where keys are variable IDs and values are the content of the variable
+              files.
     """
-    all_variable_files = get_variable_files(content_dir)
-    all_options = {}
+    if content_dir in _vars_content_cache:
+        return _vars_content_cache[content_dir]
 
-    for var_file in all_variable_files:
+    variables_content = {}
+
+    for var_file in get_variable_files(content_dir):
         try:
             yaml_content = open_and_macro_expand(var_file)
```

Comment:
I failed to run 'get_variable_values' outside of the content repo. I got error like:
```
Error processing file /xxx/content/linux_os/guide/auditing/auditd_configure_rules/var_audit_failure_mode.var: cannot access local variable 'filename' where it is not associated with a value
```
Seems like the error is still caused by loading jinja macros.  In ssg/constants.py:
```
JINJA_MACROS_DIRECTORY = os.path.abspath(os.path.join(os.path.dirname(os.path.dirname(
    __file__)), "shared", "macros"))
```

Reply:
You are correct. It should be fixed by https://github.com/ComplianceAsCode/content/pull/12816